encord/configs.py

```diff
@@ -172,17 +173,21 @@ def __init__(
         web_file_path: str = ENCORD_PUBLIC_PATH,
         domain: Optional[str] = None,
         requests_settings: RequestsSettings = DEFAULT_REQUESTS_SETTINGS,
+        user_agent_suffix: Optional[str] = None,
     ):
         if domain is None:
             raise RuntimeError("`domain` must be specified")
 
         self.domain = domain
+        self.user_agent_suffix = validate_user_agent_suffix(user_agent_suffix) if user_agent_suffix else None
         endpoint = domain + web_file_path
         super().__init__(endpoint, requests_settings=requests_settings)
 
-    @staticmethod
-    def _user_agent() -> str:
-        return f"encord-sdk-python/{encord_version} python/{platform.python_version()} pydantic/{pydantic_version_str}"
+    def _user_agent(self) -> str:
+        base_agent_header = (
+            f"encord-sdk-python/{encord_version} python/{platform.python_version()} pydantic/{pydantic_version_str}"
+        )
+        return base_agent_header + " " + self.user_agent_suffix if self.user_agent_suffix else base_agent_header
```

Comment:
This is kinda janky but I liked it returning the stripped version and then ensuring that we have a singular whitespace between the two. Could alternatively return it with a leading whitespace and have the None case -> "" such that we have:
```suggestion
        return base_agent_header + self.user_agent_suffix
```
but that's a bit magical and the function's behaviour is gnarly then